packages/calendar-controls/src/__test__/calendar-controls-plugin.spec.ts

```diff
@@ -143,4 +154,279 @@ describe('createCalendarControlsPlugin', () => {
       })
     })
   })
+
+  describe('Setting/Getting the first day of the week', () => {
+    it('should return the first day of the week that was set', () => {
+      const controlsPlugin = createCalendarControlsPlugin()
+      const $app = __createAppWithViews__({
+        plugins: [controlsPlugin],
+        selectedDate: '2021-01-01',
+        defaultView: InternalViewName.Week,
+        firstDayOfWeek: WeekDay.MONDAY,
+      })
+      controlsPlugin.init($app)
+      controlsPlugin.setFirstDayOfWeek(WeekDay.FRIDAY)
+
+      expect(controlsPlugin.getFirstDayOfWeek()).toEqual(WeekDay.FRIDAY)
+      expect($app.config.firstDayOfWeek.value).toEqual(WeekDay.FRIDAY)
+    })
+  })
+
+  describe('Setting/Getting the locale', () => {
+    it('should return the locale that was set', () => {
+      const controlsPlugin = createCalendarControlsPlugin()
+      const $app = __createAppWithViews__({
+        plugins: [controlsPlugin],
+        selectedDate: '2021-01-01',
+        defaultView: InternalViewName.Week,
+        locale: 'en-US',
+      })
+      controlsPlugin.init($app)
+
+      expect(controlsPlugin.getLocale()).toEqual('en-US')
+
+      controlsPlugin.setLocale('de-DE')
+      expect(controlsPlugin.getLocale()).toEqual('de-DE')
+      expect($app.config.locale.value).toEqual('de-DE')
+
+      // The following expectation should work, however the translate function of $app does not use the locale signal of the config
+      // expect($app.translate('Today')).toEqual('Heute')
```

Comment:
@tomosterlund 
This currently doesn't work as the `translate` function in `packages/translations/src/translator/translate.ts` does not accept the locale as a signal.

There are a couple of options that could work but we did not touch it as we didn't know what would be prefered and didn't want to break the existing interface.

It's probably a good idea to create a follow-up issue here and tackle this at a later point in time. You can see what is or isn't translated when changing locales in the development calendar.

Reply:
Yea wasn't so straightforward to make the locale update everywhere. I added a commit where I believe I managed to make the locale update work though.

I think for a few more of the now reactive config options, there might be gotchas to look out for, where the UI might not update just because the signal it depends on does. I'll continue testing a little over the next few days.

Also it'd be nice to have a couple of beta testers before releasing this, since these changes have quite the large impact. Are you guys in production yet?